libvast/vast/segment.hpp

```diff
@@ -33,7 +34,7 @@ class segment {
   /// Constructs a segment.
   /// @param header The header of the segment.
   /// @param chunk The chunk holding the segment data.
-  static caf::expected<segment> make(chunk_ptr chunk);
+  static caf::expected<segment> make(const chunk_ptr& chunk);
```

Comment:
Is it possible to use `chunk_ptr&&` here and `flatbuffer<fbs::Segment>&&` below?

I general I'm wondering if it would be worthwhile to distinguish between owned and shared flatbuffers. I guess will come up when we want to add support to mutable flatbuffers.

Also, the comment mentions a non-existing parameter `header`.

Reply:
> distinguish between owned and shared flatbuffers.

`x.chunk().unique()` does exactly that.